suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/SpdzRoundSynchronization.java

```diff
@@ -20,7 +21,12 @@
   private static final int macCheckThreshold = 100000;
   private int gatesEvaluated = 0;
   private boolean doMacCheck = false;
+  private final SecureRandom secRand;
 
+  public SpdzRoundSynchronization(SecureRandom rand) {
```

Comment:
SpdzMacCheckProtocol can create it's own secure random - why introduce a parameter here?
... there may be a good reason, however it is not presented here

Reply:
No good reason other than I like the fact that the SpdzProtocolSuite is responsible for creating the stuff that the suite needs. This includes the SecureRandom source.